### PR TITLE
[NEAT-573] 🤗Move out Validation

### PR DESCRIPTION
**Pull request by doctrino**

### CODE REVIEW

File: cognite/neat/_rules/models/information/_rules.py [hunk 1]
```diff
@@ -243,41 +234,6 @@ def parse_str(cls, values: Any) -> Any:
...
-        # update implements
-        for class_ in self.classes:
-            if class_.implements:
-                for parent in class_.implements:
-                    if not isinstance(parent.prefix, str):
-                        parent.prefix = self.metadata.prefix
-            if class_.class_.prefix is Undefined:
-                class_.class_.prefix = self.metadata.prefix
-
-        return self
```
Comment (doctrino) [inline cognite/neat/_rules/models/information/_rules.py:268]: This logic is done in the `InformationInputRules`.

File: cognite/neat/_client/_api/data_modeling_loaders.py [hunk 2]
```diff
@@ -441,6 +449,68 @@ def _delete(self, ids: SequenceNotStr[ContainerId]) -> list[ContainerId]:
     def _create_list(self, items: Sequence[Container]) -> ContainerList:
         return ContainerList(items)
 
+    def _retrieve_recursive(self, container_ids: SequenceNotStr[ContainerId]) -> ContainerList:
```
Comment (nikokaoja) [inline cognite/neat/_client/_api/data_modeling_loaders.py:452]: great again :) moving where it is suppose to be :) with increase functionality

File: cognite/neat/_client/_api/schema.py [hunk 3]
```diff
@@ -48,3 +50,72 @@ def retrieve(
             views=views,
             containers=ContainerApplyDict(containers.as_write()),
         )
+
+    def retrieve_data_model_id(self, data_model_id: dm.DataModelIdentifier) -> DMSSchema:
+        data_models = self._client.data_modeling.data_models.retrieve(data_model_id, inline_views=True)
+        if len(data_models) == 0:
+            raise NeatValueError(f"Data model {data_model_id} not found")
+        data_model = data_models.latest_version()
```
Comment (nikokaoja) [inline cognite/neat/_client/_api/schema.py:58]: Uh great ! there is method for getting latest version ! I was not aware it exists

File: cognite/neat/_client/_api/schema.py [hunk 4]
```diff
@@ -48,3 +50,72 @@ def retrieve(
...
+        if len(space_read) != len(space_ids):
+            raise NeatValueError(f"Space(s) {space_read} not found")
+        space_write = space_read.as_write()
+
+        existing_view_ids = set(views.as_ids())
+        views = self._client.loaders.views.retrieve(
+            list(existing_view_ids), include_connected=True, include_ancestor=True
+        )
+
+        missing = existing_view_ids - set(views.as_ids())
```
Comment (nikokaoja) [inline cognite/neat/_client/_api/schema.py:100]: how can you have missing? what is the edge case? I would expect you will get same views back + any views that are implemented.

File: cognite/neat/_client/_api/schema.py [hunk 5]
```diff
@@ -48,3 +50,72 @@ def retrieve(
...
+        containers = self._client.loaders.containers.retrieve(list(container_ids), include_connected=True)
+
+        space_ids = [data_model.space]
+        space_read = self._client.loaders.spaces.retrieve(space_ids)
+        if len(space_read) != len(space_ids):
+            raise NeatValueError(f"Space(s) {space_read} not found")
+        space_write = space_read.as_write()
+
+        existing_view_ids = set(views.as_ids())
+        views = self._client.loaders.views.retrieve(
```
Comment (nikokaoja) [inline cognite/neat/_client/_api/schema.py:96]: maybe rename a bit vars as `views` is assigned several times and can contain different "views"

L80 `views` is only views which are explicitly part of data model
L95 `existing_view_ids` is convience arg that you pass in
L96 where you re-assign `views` to contain the full blown set of views

File: cognite/neat/_issues/_base.py [hunk 6]
```diff
@@ -511,13 +523,10 @@ def catch_issues(
...
             yield future_result
         except ValidationError as e:
-            issues.extend(error_cls.from_pydantic_errors(e.errors(), **(error_args or {})))
-            future_result._result = "failure"
-        except MultiValueError as e:
-            issues.extend(e.errors)
+            issues.extend(error_cls.from_errors(e.errors(), **(error_args or {})))  # type: ignore[arg-type]
             future_result._result = "failure"
-        except NeatError as e:
-            issues.append(e)
```
Comment (nikokaoja) [inline cognite/neat/_issues/_base.py:520]: GREAT !

File: cognite/neat/_issues/warnings/user_modeling.py [hunk 7]
```diff
@@ -89,15 +89,15 @@ class HasDataFilterOnViewWithReferencesWarning(UserModelingWarning):
...
+class ContainerPropertyLimitWarning(UserModelingWarning):
     """The number of properties in the {view_id} view is {count} which
     is more than the API limit {limit} properties.
     This can lead to performance issues.
     Reduce the number of properties in the view."""
 
     fix = "Reduce the number of properties in the view"
 
-    view_id: ViewId
+    view_id: ContainerId
```
Comment (nikokaoja) [inline cognite/neat/_issues/warnings/user_modeling.py:100]: should this be called `container_id` ?

File: cognite/neat/_rules/exporters/_rules2dms.py [hunk 8]
```diff
@@ -39,8 +39,6 @@ class DMSExporter(CDFExporter[DMSRules, DMSSchema]):
             If set, only export components in the given spaces. Defaults to None which means all spaces.
         existing_handling (Literal["fail", "skip", "update", "force"], optional): How to handle existing components.
             Defaults to "update". See below for details.
-        export_pipeline (bool, optional): Whether to export the pipeline. Defaults to False. This means setting
```
Comment (nikokaoja) [inline cognite/neat/_rules/exporters/_rules2dms.py:42]: cleansing :)

**Review by nikokaoja — APPROVED** [review]

**Pull request merged**